crates/turbo-tasks-memory/src/task.rs

```diff
@@ -680,6 +683,11 @@ impl Task {
                     if active {
                         backend.increase_scope_active(root, turbo_tasks);
                     }
+                    if parent {
+                        backend.with_scope(root, |parent| {
+                            parent.add_parent(id, backend);
```

Comment:
This parent child naming has confused me a few times:

1. `|parent|` here is the task scope of `root`, right? Could we call it `root_scope` or something?
2. How does `root` gain a parent, isn't it the root? Shouldn't it get `children`?
3. `parent.add_parent()` is confusing, but that's tied to the naming of `parent`.